core/src/main/kotlin/ch/ergon/dope/resolvable/clause/model/JoinClause.kt

```diff
@@ -113,16 +113,23 @@ sealed class SelectJoinClause : ISelectJoinClause {
             is AliasedBucket -> joinable.asBucketDefinition().toDopeQuery(manager)
             else -> joinable.toDopeQuery(manager)
         }
-        val joinHints = if (hashOrNestedLoopHint != null || keysOrIndexHint != null) {
-            "USE" +
-                hashOrNestedLoopHint?.let { " ${hashOrNestedLoopHint.queryString}" }.orEmpty() +
-                keysOrIndexHint?.let { " ${keysOrIndexHint.toDopeQuery(manager).queryString}" }.orEmpty()
+        val hashOrNestedLoopHintDopeQuery = hashOrNestedLoopHint?.toDopeQuery(manager)
+        val keysOrIndexHintDopeQuery = keysOrIndexHint?.toDopeQuery(manager)
```

Comment:
would move these two statements inside of the following if-statement

Reply:
done